common/app/views/fragments/javaScriptLaterSteps.scala.html

```diff
@@ -334,6 +334,57 @@
 
                 });
 
+                @for(membershipAccess <- item.membershipAccess if item.requiresMembershipAccess) {
+                    /**
+                     * Membership access
+                     * Items with either of the following fields require Membership access
+                     * - membershipAccess=members-only
+                     * - membershipAccess=paid-members-only
+                     */
+                    (function(isModern) {
+
+                        /**
+                         * Authenticating requires CORS and withCredentials
+                         * If we don't cut the mustard then pass through.
+                         */
+                        if(!isModern) { return; }
+
+                        require([
+                            'common/utils/$',
+                            'common/utils/ajax'
+                        ], function($, ajax) {
+
+                            var membershipUrl = "@Configuration.id.membershipUrl",
+                                membershipAccess = "@membershipAccess",
+                                requiresPaidTier = (membershipAccess.indexOf("paid-members-only") !== -1),
+                                membershipAuthUrl = membershipUrl + '/choose-tier?membershipAccess=' + membershipAccess;
+
+                            ajax({
+                                url: membershipUrl + "/user/me",
+                                type: 'json',
+                                crossOrigin: true,
+                                withCredentials: true
+                            }).then(function (resp) {
+                                /**
+                                 * Check the users access matches the content
+                                 */
+                                var canViewContent = (requiresPaidTier) ? !!resp.tier && resp.isPaidTier : !!resp.tier;
+                                if (canViewContent) {
+                                    $('body').removeClass('has-membership-access-requirement');
+                                } else {
+                                    window.location.href = membershipAuthUrl;
+                                }
+                            }).fail(function() {
+                                /**
+                                 * If the request fails assume non-member
+                                 */
+                                window.location.href = membershipAuthUrl;
+                            });
+
+                        });
+                    }(guardian.isModernBrowser));
+                }
+
```

Comment:
Ping @rich-nguyen you are going to need to fix the conflicts this is going to create in your SystemJS branch :stuck_out_tongue_winking_eye: